### ISSUE TITLE
don't throw error if a line with no spaces is too long

### ISSUE THREAD

**Issue by YoniChechik** [charliermarsh/ruff#3460]
Here is an example code that can throw an error:
```python
# NOTE this 2 lines are a hack to make numba work:
# https://numba.readthedocs.io/en/stable/user/troubleshoot.html#my-code-has-an-untyped-list-problem
```
The code in the second line can't be trimmed down but ruff still outputs a bug of "line too long"

**Comment by charliermarsh** [comment]
I thought we _did_ allow this, I'll take a look.

**Comment by charliermarsh** [comment]
So I'm not able to reproduce an E501 error on that code. `ruff --isolated --select E501 foo.py` produces no output. (If I add some spaces within the URL, I _do_ get an error as expected.) Are you able to reproduce?

**Comment by YoniChechik** [comment]
Correct it was some other problem on my side.

Thnaks!
